src/index.js

```diff
@@ -59,7 +58,15 @@ export default function createCopyOnWriteState(baseState) {
         `CopyOnWriteStoreProvider(...): There can only be a single ` +
           `instance of a provider rendered at any given time.`
       );
+
       updateState = this.updateState;
+
+      // dequeue and call requests that pushed the queue before
+      // provider mounted
+      while (mutateQueue.length > 0) {
```

Comment:
Actually batching multiple mutate requests and calling updateState once will be more effective. I'll give a try.